File: back-end/src/main/java/nl/hva/backend/domain/models/game/board/location/locationImpl/BoardCorner.java
package nl.hva.backend.domain.models.game.board.location.locationImpl;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import nl.hva.backend.domain.models.game.board.GameBoard;
import nl.hva.backend.domain.models.game.board.location.Location;

import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;

@NoArgsConstructor
@Getter
@Setter
// JPA
@Entity
public class BoardCorner extends Location {
    public final static int DEFAULT_GO_LOCATION = 1;
    public final static int DEFAULT_JAIL_LOCATION = 11;
    public final static int DEFAULT_FREE_PARKING_LOCATION = 21;
    public final static int DEFAULT_GO_TO_JAIL_LOCATION = 31;

    @Enumerated(EnumType.STRING)
    private BoardCornerType type;

    public BoardCorner(
            long id,
            int position,
            String description,
            GameBoard gameBoard,
            BoardCornerType type
    ) {
        super(id, position, description, gameBoard);
        this.type = type;
    }

    public enum BoardCornerType {
        FREE_PARKING,
        GO,
        GO_TO_JAIL,
        JAIL
    }

}

File: front-end/src/app/domain/models/game/board/location/locationImpl/BoardCorner.ts
import {Location} from "../Location";

export enum BoardCornerType {
  FREE_PARKING = "FREE_PARKING",
  GO = "GO",
  GO_TO_JAIL = "GO_TO_JAIL",
  JAIL = "JAIL"
}

export class BoardCorner extends Location {
  public static readonly DEFAULT_GO_LOCATION: number = 1
  public static readonly DEFAULT_JAIL_LOCATION: number = 11
  public static readonly DEFAULT_FREE_PARKING_LOCATION: number = 21
  public static readonly DEFAULT_GO_TO_JAIL_LOCATION: number = 31

  private _type: BoardCornerType;

  public constructor(
    id: number,
    position: number,
    description: string,
    type: BoardCornerType
  ) {
    super(BoardCorner.name, id, position, description);
    this._type = type;
  }

  public static trueCopy(location: BoardCorner): BoardCorner | null {
    return new BoardCorner(
      location.id,
      location.position,
      location.description,
      location.type,
    );
  }

  public override toJson(): object {
    return Object.assign(
      super.toJson(),
      {
        "type": this.type,
      }
    );
  }

  public get type(): BoardCornerType {
    return this._type;
  }

  public override toString(): string {
    switch (this.type) {
      case BoardCornerType.FREE_PARKING:
        return "Free parking"
      case BoardCornerType.JAIL:
        return "Jail"
      case BoardCornerType.GO_TO_JAIL:
        return "Go to jail"
      case BoardCornerType.GO:
        return "Go"
      default:
        return "Board corner";
    }
  }

}
